server/models/services/CollectionJobService.js

```diff
@@ -329,19 +330,23 @@ const scheduleCollectionJob = async (
         jobId = '1';
     }
 
-    const automationSchedulerResponse = await axios.post(
-        `${process.env.AUTOMATION_SCHEDULER_URL}/jobs/new`,
-        {
-            testPlanVersionGitSha: gitSha,
-            testIds: testIds ?? tests.map(t => t.id),
-            testPlanName: directory,
-            jobId
-        },
-        axiosConfig
-    );
+    let status = 'QUEUED';
+
+    if (!process.env.AUTOMATION_CALLBACK_FQDN) {
```

Comment:
I'd rather use a more semantic condition to make the code a little more self-documenting.

```suggestion
    if (process.env.ENVIRONMENT !== 'production') {
```

Reply:
I had initially thought of this approach, but that setup would make testing in development branch harder, which is why I used an environment variable that could act as a toggle regardless of production/dev/testing "node envrionment" - I am happy to make it a more obvious one like ENABLE_AUTOMATION_JOB_CREATE or something similar (not sold on the name, feel free to paint the shed)

Reply:
AUTOMATION_JOB_SCHEDULER=production|mock

Reply:
I'm generally averse to complexities in the definition of the deployment environment. Specifically in this case, that means creating a two-dimensional space rather than the current enumeration. Any complexity complicates maintenance (since contributors have to perform some mental combinatorics to determine if/when their work needs to be modulated), and the introduction of an independent variable often creates states which aren't needed. All of these things also pose a hazard during deployment since they introduce new opportunities for misconfiguration.

[We were considering a similar change earlier this year, but we managed to find a solution that minimized the configuration space.](https://github.com/w3c/aria-at-app/pull/492#issuecomment-1447395777) I don't know if that remediation is appropriate here, but it does motivate me to ask: what kind of difficulties arise from using the existing `ENVIRONMENT` environment variable for this purpose?

Reply:
I tend to agree with the points and the linked issue. My reluctance is more about not wanting to have to set my local server up in "production" to be able to test full cycle communications - I think 95% of the time local development environment will want to mock these jobs, but for the situations in which we do want to work directly on local (like working on this PR or future changes to these services), switching my local dev environment to "production" mode has other possible unknown (to me atm) behavior switches.  If you don't think there are any reasons to fear this step for local testing of GitHub agent job creation then I totally agree with the assessment.